workers/autopruneworker.py

```diff
@@ -35,21 +37,36 @@ def prune(self):
                 autoprune_task.id,
                 autoprune_task.namespace,
             )
-
+            repo_policies = []
             try:
-                policies = get_namespace_autoprune_policies_by_id(autoprune_task.namespace)
-                if not policies:
-                    # When implementing repo policies, fetch repo policies before deleting the task
-                    delete_autoprune_task(autoprune_task)
-                    continue
+                ns_policies = get_namespace_autoprune_policies_by_id(autoprune_task.namespace)
+                if not ns_policies:
+                    repo_policies = get_repository_autoprune_policies_by_namespace_id(
+                        autoprune_task.namespace
+                    )
+                    if not repo_policies:
+                        delete_autoprune_task(autoprune_task)
+                        continue
 
                 execute_namespace_polices(
-                    policies,
+                    ns_policies,
                     autoprune_task.namespace,
                     FETCH_REPOSITORIES_PAGE_LIMIT,
                     FETCH_TAGS_PAGE_LIMIT,
                 )
 
+                # case: only repo policies exists & no namespace policy
+                for policy in repo_policies:
+                    repo_id = policy.repository_id
+                    repo = get_repository_by_policy_repo_id(repo_id)
+                    logger.info(
+                        "processing autoprune task %s for repository %s",
+                        repo.name,
+                    )
+                    execute_policy_on_repo(
+                        policy, repo_id, autoprune_task.namespace, tag_page_limit=100
+                    )
+
```

Comment:
The reason I added within the context of `ns_policy` execution [here](https://github.com/quay/quay/pull/2691/files#diff-0445600d80e07f49e4ac5d97b6853caebdeec7024679044c9c62c3b8eb31154bR613-R616) is to allow for future code optimizations and/or support new policy methods. Having all policies funneled at one place would allow us to reconcile policies before applying them. I'll keep this for now and follow up with another PR if we decide to change this bit.

Reply:
sounds good